tools/distpartitioning/dist_lookup.py

```diff
@@ -136,107 +138,134 @@ def get_partition_ids(self, global_nids):
-        # the final result (partition-ids) in the same order as the global-nids (function argument)
-        send_list = []
-        indices_list = []
-        for idx in range(self.world_size):
-            idxes = np.where(service_owners == idx)
-            ll = global_nids[idxes[0]]
-            send_list.append(torch.from_numpy(ll))
-            indices_list.append(idxes[0])
-        assert len(np.concatenate(indices_list)) == len(global_nids)
-        assert np.all(np.sort(np.concatenate(indices_list)) == np.arange(len(global_nids)))
-
-        # Send the request to everyone else.
-        # As a result of this operation, the current process also receives a list of lists
-        # from all the other processes. 
-        # These lists are global-node-ids whose global-node-ids <-> partition-id mappings 
-        # are owned/stored by the current process
-        owner_req_list = alltoallv_cpu(self.rank, self.world_size, send_list)
-
-        # Create the response list here for each of the request list received in the previous
-        # step. Populate the respective partition-ids in this response lists appropriately
-        out_list = []
-        for idx in range(self.world_size):
-            if owner_req_list[idx] is None:
-                out_list.append(torch.empty((0,), dtype=torch.int64))
-                continue
-            # Get the node_type_ids and per_type_nids for the incoming global_nids.
-            ntype_ids, type_nids = self.id_map(owner_req_list[idx].numpy())
-            ntype_ids, type_nids = ntype_ids.numpy(), type_nids.numpy()
-
-            # Lists to store partition-ids for the incoming global-nids.
-            type_id_lookups = []
-            local_order_idx = []
-
-            # Now iterate over all the node_types and acculumulate all the partition-ids
-            # since all the partition-ids are based on the node_type order... they
-            # must be re-ordered as per the order of the input, which may be different.
-            for tid in range(len(self.partid_list)):
-                cond = ntype_ids == tid
-                local_order_idx.append(np.where(cond)[0])
-                global_type_nids = type_nids[cond]
-                if len(global_type_nids) <= 0:
-                    continue
-
-                local_type_nids = global_type_nids - self.type_nid_begin[tid]
-
-                assert np.all(local_type_nids >= 0)
-                assert np.all(local_type_nids <= (self.type_nid_end[tid] + 1 - self.type_nid_begin[tid]))
+        CHUNK_SIZE = 200 * 1000 * 1000
+        # Determine the no. of times each process has to send alltoall messages.
+        local_rows = agg_global_nids.shape[0]
+        all_sizes = allgather_sizes([local_rows], self.world_size, self.num_parts, return_sizes=True)
+        max_count = np.amax(all_sizes)
+        num_splits = (max_count // CHUNK_SIZE) + (0 if (max_count % CHUNK_SIZE == 0) else 1)
```

Comment:
`ceil()`?

Reply:
Agreed with the suggestion. Implemented this change per suggestion.